logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/UserSessionScope.kt

```diff
@@ -115,9 +128,18 @@ abstract class UserSessionScopeCommon(
             eventMapper
         )
 
-    val listenToEvents: ListenToEventsUseCase get() = ListenToEventsUseCase(syncManager, eventRepository)
+    protected abstract val protoContentMapper: ProtoContentMapper
+    private val conversationEventReceiver: ConversationEventReceiver
+        get() = ConversationEventReceiver(
+            authenticatedDataSourceSet.proteusClient,
+            messageRepository,
+            protoContentMapper
+        )
+    val listenToEvents: ListenToEventsUseCase get() = ListenToEventsUseCase(syncManager, eventRepository, conversationEventReceiver)
     val client: ClientScope get() = ClientScope(clientRepository, authenticatedDataSourceSet.proteusClient)
     val conversations: ConversationScope get() = ConversationScope(conversationRepository, syncManager)
-    val messages: MessageScope get() = MessageScope(messageRepository)
+
+    //TODO How to get self domain?
```

Comment:
From the self user

Reply:
Yes!
I changed it to use the repository now instead of relying on a fixed initial value.